common/djangoapps/student/views.py

```diff
@@ -707,10 +810,54 @@ def dashboard(request):
         'order_history_list': order_history_list,
         'courses_requirements_not_met': courses_requirements_not_met,
         'nav_hidden': True,
-        'course_programs': course_programs,
+        'course_programs': course_programs
+    }
+
+    return render_to_response('dashboard.html', context, content_type='application/json')
+
+
+def _get_courses_requirements(course_requirements):
+    """ Get dict of course_requirements and convert CourseLocator to unicode.
+
+    Arguments:
+        course_requirements (list): a list of course requirements.
+
+    Returns:
+        Return dict contained list of course requirements.
+
+    """
+    if not course_requirements:
+        return
+
+    return {
+        'courses': [
+            {
+                'key': unicode(course['key']),
+                'display': course['display']
+            } for course in course_requirements.get('courses')
+        ]
     }
 
-    return render_to_response('dashboard.html', context)
+
+def _get_share_config(request, course_overview):
+    """
+    Get sharing configuration for a course.
+    """
+    share_url = ''
+    if settings.SOCIAL_SHARING_SETTINGS.get("CUSTOM_COURSE_URLS", False):
+        if course_overview.social_sharing_url:
+            share_url = urllib.quote_plus(course_overview.social_sharing_url)
+    else:
+        share_url = urllib.quote_plus(
+            request.build_absolute_uri(reverse('about_course', args=[unicode(course_overview.id)]))
+        )
+
+    return {
+        "share_url": share_url,
+        "share_window_name": "shareWindow",
```

Comment:
Move this to the template where it's used. There is no need to pass hard-coded text to the templates for every enrollment.